File: 36-Extra-Challenges/115-Includes.py
def includes(collection, search_value, start_index=0):
    if type(collection) == dict:
        return search_value in collection.values()
    else:
        return search_value in collection[start_index:]


print(includes([1, 2, 3], 1)) # True 
print(includes([1, 2, 3], 1, 2)) # False 
print(includes({ 'a': 1, 'b': 2 }, 1)) # True 
print(includes({ 'a': 1, 'b': 2 }, 'a')) # False
print(includes('abcd', 'b')) # True
print(includes('abcd', 'e')) # False

File: 36-Extra-Challenges/115-Includes.js
const myIncludes = (collection, searchValue, startIndex=0) => {
    if (typeof collection === 'string' || Array.isArray(collection)) {
        return collection.slice(startIndex).includes(searchValue)
    }
    else {
        return Object.values(collection).includes(searchValue)
    }
}


console.log(myIncludes([1, 2, 3], 1)) // True 
console.log(myIncludes([1, 2, 3], 1, 2)) // False 
console.log(myIncludes({ 'a': 1, 'b': 2 }, 1)) // True 
console.log(myIncludes({ 'a': 1, 'b': 2 }, 'a')) // False
console.log(myIncludes('abcd', 'b')) // True
console.log(myIncludes('abcd', 'e')) // False